lighthouse-extension/build-extension.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+
+const archiver = require('archiver');
+const del = require('del');
+const cpy = require('cpy');
+const makeDir = require('make-dir');
+const bundleBuilder = require('./bundle-builder.js');
+
+const distDir = 'dist';
+const manifestVersion = require(`./app/manifest.json`).version;
+
+// list of all consumers we build for (easier to understand which file is used for which)
+const CONSUMERS = {
+  DEVTOOLS: {
+    src: 'devtools-entry.js',
+    dist: 'lighthouse-dt-bundle.js',
+  },
+  EXTENSION: {
+    src: 'extension-entry.js',
+    dist: 'lighthouse-ext-bundle.js',
+  },
+  LIGHTRIDER: {
+    src: 'lightrider-entry.js',
+    dist: 'lighthouse-lr-bundle.js',
+  },
+};
+
+/**
+ * Browserify and minify scripts.
+ */
+function buildAll() {
+  return Object.values(CONSUMERS).map(consumer => {
+    const inFile = `app/src/${consumer.src}`;
+    const outFile = `dist/scripts/${consumer.dist}`;
+    return bundleBuilder.build(inFile, outFile);
+  });
+}
+
+/**
+ * Copy popup.js to dist folder, inlining the current commit hash along the way.
+ * @return {Promise<void>}
+ */
+async function copyPopup() {
+  let popupSrc = fs.readFileSync('app/src/popup.js', {encoding: 'utf8'});
+  popupSrc = popupSrc.replace(/__COMMITHASH__/g, bundleBuilder.COMMIT_HASH);
+
+  await makeDir(`${distDir}/scripts`);
+  fs.writeFileSync(`${distDir}/scripts/popup.js`, popupSrc);
+}
+
+/**
+ * @return {Promise<void>}
+ */
+async function copyAssets() {
+  await cpy('app/*.html', distDir);
```

Comment:
Promise.all might be faster for all these tasks?

Reply:
it doesn't appear to make much of a difference (around 50ms either way on my machine). I imagine things are disk-bound at that point, since it's browserifying (which touches a whoooole lot of files) at the same time

Reply:
yeah indeed

Reply:
buuut, no reason not to do the globbing and copying all in one call (and in theory, cpy will handle load balancing the streams). Fixed